bin/spice/cmd/login.go

```diff
@@ -227,6 +229,27 @@ spice login snowflake --account <account-identifier> --username <username> --pri
 	},
 }
 
+var m365Cmd = &cobra.Command{
+	Use:   "m365",
+	Short: "Login to a Microsoft 365 account",
+	Example: `
+	spice login m365 --username <username> --password <password> --tenant-id <tenant-id> --client-id <client-id>
+
+# See more at: https://docs.spiceai.org/
```

Comment:
Basic auth could be a good start for initial implementation, but username/password auth is a legacy/not recommended option so we should add other auth options support as well IMO